superset/db_engine_specs/ocient.py

```diff
@@ -215,32 +222,55 @@ class OcientEngineSpec(BaseEngineSpec):
 
     @classmethod
     def get_table_names(
-        cls, database: "Database", inspector: Inspector, schema: Optional[str]
+        cls, database: Database, inspector: Inspector, schema: Optional[str]
     ) -> List[str]:
         return sorted(inspector.get_table_names(schema))
 
     @classmethod
     def fetch_data(cls, cursor, lim=None):
         rows = super(OcientEngineSpec, cls).fetch_data(cursor)
```

Comment:
What happens if an exception is raised from `fetch_data(...)`?

Reply:
Superset propagates it to the frontend in a big red box

Reply:
Additionally, if there are no rows, superset will output "The query returned no data"

Reply:
and what happens to your cached entry?

Reply:
Removed

Reply:
even if this line here raises an exception? Where does that exception get handled?
```
super(OcientEngineSpec, cls).fetch_data(cursor)
```

Reply:
Yes, errors here will also be propagated to the web UI

Reply:
Let me restate my question a little more clearly. My question is _how_ does any line after 236 in `fetch_data(...)` run if line 236 raises an exception and there isn't a try-catch block to handle the exception?

Reply:
Try-catch added to delete from cache when erroring